src/test/java/pl/piekoszek/gorskimatches/equation/SolutionToQuizzesMapperTest.java

```diff
@@ -0,0 +1,38 @@
+package pl.piekoszek.gorskimatches.equation;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SolutionToQuizzesMapperTest {
+
+    private final SolutionToQuizzesMapper solutionToQuizzesMapper = new SolutionToQuizzesMapper();
+
+    @Test
+    void shouldCreateQuizzesFromSolutionByMovingMatchInsideOneCharacter() {
+        assertThat(solutionToQuizzesMapper.insideSingleMatch("6+1=7"))
+                .containsKeys("9+1=7");
```

Comment:
in this case, 0+1=7 is also possible

Reply:
Also this test is not complete. It only checks if key it present in map, no assurance about value, I'd suggest:
```
var quizSolutionMap = solutionToQuizzesMapper.insideSingleMatch("6+1=7");

assertThat(quizSolutionMap.get("0+1=7")).containsExactly("6+1=7");
assertThat(quizSolutionMap.get("9+1=7")).containsExactly("6+1=7");
assertThat(quizSolutionMap).hasSize(2);
        ```